lib/utils.js

```diff
@@ -20,7 +20,15 @@ const labelParser = function (labels) {
 }
 /* Fingerprinting */
 const shortHash = require('short-hash')
-const fingerPrint = function (text, hex) {
+const xxh = require('xxhashjs')
+const xxhSeed = 'cloki'
+const fingerPrint = function (text, hex, alg) {
+  alg = alg || process.env.HASH || 'short-hash'
+  switch (alg) {
+    case 'XXH':
+      if (hex) return xxh.h64(text, xxhSeed).toString(16)
+      else return Number(xxh.h64(text, xxhSeed))
```

Comment:
`Number(UInt64)` seems to lose data:
```
> xxh.h64(0).update('test').digest().toString(10)
'5754696928334414137'
> Number(xxh.h64(0).update('test').digest())
5754696928334414000
```

Not sure (I am bad at javascript) - maybe a string can be returned here?

Reply:
Tests do pass with this (and forced XXH by default):

```suggestion
      else return xxh.h64(text, '').toString(10)
```

probably because inserting numbers as strings works for numeric columns in clickhouse. this way all 64 bits can be used.